lib/widgets/CustomButton/custom_button.dart

```diff
@@ -0,0 +1,51 @@
+import 'package:flutter/material.dart';
+import 'package:classroom_app/styles/color/colors.dart';
+
+class CustomButton extends StatelessWidget {
+  const CustomButton({
+    super.key,
+    required this.buttonText,
+    required this.buttonBgColor,
+    this.buttonTextColor,
+    this.buttonBorderColor,
+  });
+
+  final String buttonText;
+  final Color buttonBgColor;
+  final Color? buttonTextColor;
+  final Color? buttonBorderColor;
+
+  @override
+  Widget build(BuildContext context) {
+    double screenWidth = MediaQuery.of(context).size.width;
+    return SizedBox(
+      height: 56,
+      width: screenWidth * 0.4,
+      child: ElevatedButton(
+        onPressed: () {},
+        style: ElevatedButton.styleFrom(
+          padding: EdgeInsets.symmetric(
+            horizontal: screenWidth * 0.08,
+            vertical: 14,
+          ),
+          backgroundColor: buttonBgColor,
```

Comment:
Hardcoding colors and styles can make the button less consistent with the app’s theme.
Use Theme.of(context) to derive styles, allowing the button to adapt to global theme settings.

```suggestion
          backgroundColor: buttonBgColor ?? Theme.of(context).primaryColor,
```

Reply:
![Screenshot from 2024-12-05 12-10-49](https://github.com/user-attachments/assets/1e82f53b-f486-4b98-9296-9cfbe2d73bb8)

So, 'Theme.of(context).primary' will never be executed.